### PR TITLE
add arbitrary proxy

### PR DESCRIPTION
**Pull request by patrickkettner**
this allows for us to get around things like CORS requests

### CODE REVIEW

File: lib/openProxy.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+var http = require('http'),
+    url = require('url');
+
+module.exports = function(req, res, next) {
+  var proxiedUrl = {};
+  try {
+    proxiedUrl = url.parse(url.parse(req.url).path.match(/^\/proxy\/(.*)/));
+  } catch (err) {}
+
+  if (proxiedUrl) {
```
Comment (jhudson8) [inline lib/openProxy.js:10]: if you are going to do the protocol check, I'd make that a condition for proxying
`if (proxiedUrl && proxiedUrl.protocol.indexOf('http') === 0)`

File: lib/openProxy.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+var http = require('http'),
+    url = require('url');
+
+module.exports = function(req, res, next) {
+  var proxiedUrl = {};
+  try {
+    proxiedUrl = url.parse(url.parse(req.url).path.match(/^\/proxy\/(.*)/));
```
Comment (jhudson8) [inline lib/openProxy.js:7]: how do these proxy URLs get created in the first place?

Comment (patrickkettner) [inline lib/openProxy.js:7]: [like this](https://github.com/walmartlabs/us-mweb/pull/1418/files#r9501061)

File: lib/openProxy.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+var http = require('http'),
+    url = require('url');
+
+module.exports = function(req, res, next) {
```
Comment (kpdecker) [inline lib/openProxy.js:4]: What does this do?

File: lib/openProxy.js (hunk 1 above)
Comment (patrickkettner) [inline lib/openProxy.js:10]: only issue is that `proxiedUrl.protocol` can be null. like if you `url.parse` '/proxy/foo'

**Comment by kpdecker** [comment]
I don't like opening an endpoint that any one can hit. It's too prone for mischief that I'd like to avoid, even if it's only a test environment. Closing now but we can discuss the use case to see if this is the only solution.

**Comment by patrickkettner** [comment]
@kpdecker - How would you suggest we strip out CORS headers for the API stuff?

**Comment by kpdecker** [comment]
We dont. We need those there for an accurate test. We will have to run from a domain that they support. Baring that we create something specifically in our server tier that proxies.

To support proper CORS proxying with a solution like this we would need to forward cookies on to the proxied service and since that can be anything that creates a massive security hole that we both can't do internally and I don't want to create such a footgun in an OSS project because of how easy it would be for that to be used incorrectly and cause massive problems if found.

**Comment by patrickkettner** [comment]
Fair enough.

Have some other questions I'll ping you with in chat.

**Pull request closed without merging**